### PR TITLE
Make GitHub Actions run tests on Linux and Windows, adjust tests for Linux

### PR DESCRIPTION
**Pull request by gdziadkiewicz**
### Why
@misterspeedy Thank you for creating this amazing library! FsExcel allowed me to quickly create readable code generating Excel sheets and avoid unwanted exposure to low-level libs.

The only problem was that my code base was on netstandard20. I locally resolved it by copying the code and compiling my netstandard20 version ( no code changes needed) but I felt that it would be nice to share this possibility with everyone.

I quickly found out that the current GitHub Actions build does not execute tests and that current tests are not Linux-ready. As I don't feel comfortable with proposing any changes (like the netstandard21 -> netstandard20 switch) without test results here I am with this PR that resolves issues blocking GitHub Action testing on Windows and Linux.

If I get this merged I will come back with the netstandard21 -> netstandard20 change trying to convince you that it is safe based on the tests that didn't get red from the change :)

### How
1. Changed the GitHub Actions build to work on the sln file rather than fsproj of the lib project. Additionally, added Windows build and the steps executed in vs code tasks to ensure that generated files are fresh and up to date.
2.  Added a workaround for the missing Calibri font preventing the opening of Excel files on Linux (https://docs.closedxml.io/en/latest/tips/missing-font.html)
3.  Resolved `"/temp"` path issues
4.  Ignored enumerate fonts test

### CODE REVIEW

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -8,22 +8,38 @@ on:
...
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Setup .NET
-      uses: actions/setup-dotnet@v1
+      uses: actions/setup-dotnet@v3
       with:
-        dotnet-version: 6.0.x
+        dotnet-version: |
+          6.0.x
```
Comment (gdziadkiewicz) [inline .github/workflows/dotnet.yml:21]: Unit tests project and regression tests projects are running on different .NET versions hence two versions

File: .github/workflows/dotnet.yml [hunk 2]
```diff
@@ -8,22 +8,38 @@ on:
...
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Setup .NET
-      uses: actions/setup-dotnet@v1
+      uses: actions/setup-dotnet@v3
```
Comment (gdziadkiewicz) [inline .github/workflows/dotnet.yml:18]: Used the opportunity to update versions, can rollback if uncalled for

File: .github/workflows/dotnet.yml [hunk 3]
```diff
@@ -8,22 +8,38 @@ on:
...
       with:
-        dotnet-version: 6.0.x
+        dotnet-version: |
+          6.0.x
+          7.0.x
     - name: Restore dependencies
-      run: dotnet restore src/FsExcel
+      run: dotnet restore
+      working-directory: src
```
Comment (gdziadkiewicz) [inline .github/workflows/dotnet.yml:25]: The core of the change is doing the dotnet commands on the sln file in src folder. Previously only lib code was built/restored

File: .github/workflows/dotnet.yml [hunk 4]
```diff
@@ -8,22 +8,38 @@ on:
...
     - name: Build
-      run: dotnet build --no-restore src/FsExcel
+      run: dotnet build --no-restore
+      working-directory: src
+    - name: Dib to Markdown
```
Comment (gdziadkiewicz) [inline .github/workflows/dotnet.yml:29]: Added the vs code tasks to make sure that files are up to date

File: .github/workflows/dotnet.yml [hunk 5]
```diff
@@ -8,22 +8,38 @@ on:
...
+      working-directory: src/Scripts
+    - name: Create regression
+      run: dotnet fsi CreateRegressionTestActuals.fsx
+      working-directory: src/Scripts
     - name: Test
-      run: dotnet test --no-build --verbosity normal src/FsExcel
+      run: dotnet test --verbosity normal
+      working-directory: src
     - name: Publish FsExcel
+      if: matrix.os == 'ubuntu-latest'
```
Comment (gdziadkiewicz) [inline .github/workflows/dotnet.yml:42]: I didn't check how the build handles publishing from PR builds (what's blocking version publish from PR build?), but to preserve the original env used for NuGet publishing I added the if to limit the step to Ubuntu.

File: .vscode/settings.json [hunk 6]
```diff
@@ -1,4 +1,5 @@
 {
     "task.allowAutomaticTasks": "on",
-    "editor.inlayHints.enabled": "offUnlessPressed"
+    "editor.inlayHints.enabled": "offUnlessPressed",
```
Comment (gdziadkiewicz) [inline .vscode/settings.json:3]: I will remove that

File: README.md [hunk 7]
```diff
@@ -272,8 +272,6 @@ let headingStyle =
 
 You can set the font name using `FontName` and the size using `FontSize`:
 
-<!-- Test -->
```
Comment (gdziadkiewicz) [inline README.md:275]: This depends on the fonts installed on a given env. I propose to not use this as a test as it is too brittle.

File: src/Notebooks/Tutorial.dib [hunk 8]
```diff
@@ -302,8 +302,6 @@ let headingStyle =
 
 You can set the font name using `FontName` and the size using `FontSize`:
 
-<!-- Test -->
```
Comment (gdziadkiewicz) [inline src/Notebooks/Tutorial.dib:305]: This depends on the fonts installed on a given env. I propose to not use this as a test as it is too brittle.

File: src/Tests/RegressionTests/RegressionTests.fs [hunk 9]
```diff
@@ -11,14 +12,20 @@ let actualsPath = "../../../Actual"
...
-    [<Fact>]
-    member _.``RegressionTests`` () =
+    static member files =
         expectedsPath
         |> Directory.EnumerateFiles
-        |> Seq.iter (Path.GetFileName >> (Check.fromFilename output))
+        |> Seq.map (Path.GetFileName >> Array.singleton)
+
+    [<Theory>]
+    [<MemberData("files")>]
```
Comment (gdziadkiewicz) [inline src/Tests/RegressionTests/RegressionTests.fs:30]: Quality of life upgrade - with this each of the tests is displayed as a separate test case in test runners.

File: src/Tests/RegressionTests/RegressionTests.fs [hunk 10]
```diff
@@ -11,14 +12,20 @@ let actualsPath = "../../../Actual"
...
     let fromFilename(output : ITestOutputHelper) (filename : string) =
-        let expected = new XLWorkbook(Path.Combine(expectedsPath, filename))
         let actual = new XLWorkbook(Path.Combine(actualsPath, filename))
+        let expected = new XLWorkbook(Path.Combine(expectedsPath, filename))
         Assert.Workbook.Equal(expected, actual, filename, output)    
 
 type Tests(output : ITestOutputHelper) =
+    do
+        if RuntimeInformation.IsOSPlatform(OSPlatform.Linux) then
+            LoadOptions.DefaultGraphicEngine <- ClosedXML.Graphics.DefaultGraphicEngine("DejaVu Sans")
```
Comment (gdziadkiewicz) [inline src/Tests/RegressionTests/RegressionTests.fs:22]: This works around the fact that Calibri font is not available on Linux build agents

File: .github/workflows/dotnet.yml (hunk 1 above)
Comment (gdziadkiewicz) [reply .github/workflows/dotnet.yml:21]: Alternatively I can unify the runtime for the tests

File: .vscode/settings.json (hunk 6 above)
Comment (gdziadkiewicz) [reply .vscode/settings.json:3]: or not if it seems fine

**Comment by gdziadkiewicz** [comment]
(Build results on my fork can be found here https://github.com/gdziadkiewicz/FsExcel/pull/1 )

**Comment by smoothdeveloper** [comment]
@gdziadkiewicz I also want the library to target netstandard2.0 (to be useful on .NET Framework), and I faced similar issue for running the tests on macos.

I've made a PR which does only the test adjustments (#53) but without the need to remove the test with fonts, only skipping it if it is not running on windows for now.

I think rebasing this PR on top of #53 to only focus on CI and target would make sense (assuming the tests come green in your environment, without changing any .fs, .fsx, .xlsx files in this branch).

**Comment by misterspeedy** [comment]
WOW! Thank you so much @gdziadkiewicz  and @smoothdeveloper !  Sorry for the delay but I have now merged the PRs.  I didn't read your comment about rebasing until I had merged the main PR, but hopefully I have reinstated everything to what you both intended.  The test pass on Windows for me.

Thank you so much for these improvements.

**Comment by smoothdeveloper** [comment]
@misterspeedy I'm confirming that the changes in my PR are indeed effective, able to run the tests and get those green.

If you could check #54 and publish the library again, I think it should enable the .net framework users to leverage the library.

Thanks for the stylish library!

**Pull request merged**